### PR TITLE
Update Orchard commitment tree hashes to use total MerkleCRH^Orchard.

### PR DESCRIPTION
**Pull request by nuttycom**
See zcash/zips#530

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1044,7 +1044,7 @@ checksum = "624a8340c38c1b80fd549087862da4ba43e08858af025b236e509b6649fc13d5"
 [[package]]
 name = "orchard"
 version = "0.0.0"
-source = "git+https://github.com/zcash/orchard.git?rev=21b77d6ec56fc7e37aab9da68e88bc5dd754fe10#21b77d6ec56fc7e37aab9da68e88bc5dd754fe10"
+source = "git+https://github.com/nuttycom/orchard.git?rev=64c5b25d1c10d43b308ef636348962c21c1aa279#64c5b25d1c10d43b308ef636348962c21c1aa279"
```
Comment (daira) [inline Cargo.lock:1047]: Update this if https://github.com/zcash/orchard/pull/136 is merged first.

File: src/rust/include/rust/orchard/incremental_sinsemilla_tree.h [hunk 2]
```diff
@@ -61,13 +61,8 @@ bool orchard_merkle_frontier_append_bundle(
...
         const OrchardBundlePtr* bundle);
 
-// Compute the root of the provided orchard Merkle frontier
-//
-// If the bottom value is not derived by the Sinsemilla hash, `digest_ret` will
-// point to the resulting digest and this procedure returns `true`.
-// If the Sinsemilla hash returns the bottom value, `digest_ret` will be
-// unaltered and this procedure returns `false`.
-bool orchard_merkle_frontier_root(
+// Computes the root of the provided orchard Merkle frontier
```
Comment (daira) [inline src/rust/include/rust/orchard/incremental_sinsemilla_tree.h:64]:
```suggestion
// Computes the root of the provided Orchard Merkle frontier.
```

**Review by daira — APPROVED** [review]
utACK

File: Cargo.toml [hunk 3]
```diff
@@ -65,7 +65,7 @@ codegen-units = 1
 ed25519-zebra = { git = "https://github.com/ZcashFoundation/ed25519-zebra.git", rev = "d3512400227a362d08367088ffaa9bd4142a69c7" }
 halo2 = { git = "https://github.com/zcash/halo2.git", rev = "d04b532368d05b505e622f8cac4c0693574fbd93" }
 incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "8b59049f1746827ffa3763efa8af948f680491d0" }
-orchard = { git = "https://github.com/zcash/orchard.git", rev = "21b77d6ec56fc7e37aab9da68e88bc5dd754fe10" }
+orchard = { git = "https://github.com/nuttycom/orchard.git", rev = "64c5b25d1c10d43b308ef636348962c21c1aa279" }
```
Comment (str4d) [inline Cargo.toml:68]:
```suggestion
orchard = { git = "https://github.com/zcash/orchard.git", rev = "74df35ce890726478983fb892ff38a5e44ceb08a" }
```

File: src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs [hunk 4]
```diff
@@ -118,17 +118,12 @@ pub extern "C" fn orchard_merkle_frontier_root(
             .expect("Cannot return to the null pointer.")
     };
 
-    if let Some(root) = tree.root().to_bytes() {
-        root_ret.copy_from_slice(&root);
-        true
-    } else {
-        false
-    }
+    root_ret.copy_from_slice(&tree.root().to_bytes());
```
Comment (str4d) [inline src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs:121]:
```suggestion
    *root_ret = tree.root().to_bytes();
```

File: src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs [hunk 5]
```diff
@@ -278,12 +273,7 @@ pub extern "C" fn incremental_sinsemilla_tree_root(
...
```
Comment (str4d) [inline src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs:276]:
```suggestion
    *root_ret = tree.root().to_bytes();
```

File: src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs [hunk 6]
```diff
@@ -296,9 +286,7 @@ pub extern "C" fn incremental_sinsemilla_tree_empty_root(root_ret: *mut [u8; 32]
 
     let altitude = Altitude::from(MERKLE_DEPTH);
 
-    let digest = OrchardIncrementalTreeDigest::empty_root(altitude)
-        .to_bytes()
-        .unwrap();
+    let digest = MerkleCrhOrchardOutput::empty_root(altitude).to_bytes();
 
     root_ret.copy_from_slice(&digest);
```
Comment (str4d) [inline src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs:291]:
```suggestion
    *root_ret = digest;
```

**Review by str4d — APPROVED** [review]
utACK after dependency update.

**Review by daira — APPROVED** [review]
utACK

**Comment by nuttycom** [comment]
@zkbot r+

**Pull request merged**